std/bitmanip.d

```diff
@@ -72,6 +73,18 @@ private string myToString(ulong n)
     return cast(string) s ~ (n > uint.max ? "UL" : "U");
 }
 
+private template TypeName(T)
+{
+    static if(__traits(compiles, TemplateOf!T))
+    {
+        enum TypeName = T.stringof~"!"~TemplateArgsOf!T.stringof[5..$];
+    }
+    else
+    {
+        enum TypeName = T.stringof;
+    }
+}
```

Comment:
Hmm, I was just looking at `std.traits` for five minutes, because I was convinced that something such general like this already existed, but apparently it doesn't.